common-kafka/src/main/java/com/cerner/common/kafka/consumer/ProcessingKafkaConsumer.java

```diff
@@ -755,17 +755,19 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitionsAssigned)
             LOGGER.info("Rebalance [{}] is in progress. Old partitions: {} New Partitions: {}", REBALANCE_COUNTER.count(),
                     partitions.keySet(), partitionsAssigned);
 
-            // Close all partitions we will no longer use
-            partitions.values().stream()
-                    .filter(partition -> !partitionsAssigned.contains(partition.getTopicPartition()))
+            /**
+             * We must reset the local partition state (cached offsets) for all partitions,
+             * even if the last assignment we saw also included these partitions,
+             * because we may have been disconnected and missed an entire group generation
+             * during which another consumer might have been assigned the partition and committed offsets.
+             */
+            partitions.values()
                     .forEach(partition -> IOUtils.closeQuietly(partition));
 
-            // Remove all partitions we are no longer assigned to
-            partitions.keySet().retainAll(partitionsAssigned);
+            partitions.clear();
 
-            // Add all partitions that we don't already have
             try {
-                partitionsAssigned.stream().filter(tp -> !partitions.containsKey(tp))
+                partitionsAssigned.stream()
```

Comment:
Can remove the "extraneous stream" here too

Reply:
done